pkg/controller/installation/marketplace/manager.go

```diff
@@ -51,73 +54,81 @@ func NewManager() *MarketplaceManager {
 	return &MarketplaceManager{}
 }
 
-func (m *MarketplaceManager) CreateSubscription(ctx context.Context, serverClient pkgclient.Client, owner ownerutil.Owner, os marketplacev1.OperatorSource, ns string, pkg string, channel string, operatorGroupNamespaces []string, approvalStrategy coreosv1alpha1.Approval) error {
-	sub := &coreosv1alpha1.Subscription{
-		ObjectMeta: metav1.ObjectMeta{
-			Namespace: ns,
-			Name:      pkg,
-		},
-	}
+type Target struct {
+	Namespace, Pkg, Channel string
+}
 
+func (m *MarketplaceManager) createAndWaitCatalogSourceConfig(ctx context.Context, owner ownerutil.Owner, t Target, os marketplacev1.OperatorSource, client pkgclient.Client, sub *coreosv1alpha1.Subscription) (string,error) {
```

Comment:
This should be `createAndWaitCatalogSource` now. And I don't think `sub` is used.